### PR TITLE
NODE-2626 Corrected snapshot hash of LeaseState

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: node/src/main/scala/com/wavesplatform/state/TxStateSnapshotHashBuilder.scala [hunk 1]
```diff
@@ -75,7 +75,17 @@ object TxStateSnapshotHashBuilder {
     } addEntry(KeyType.AssetScript, asset.id.arr)(scriptInfo.script.bytes().arr)
 
     snapshot.leaseStates.foreach { case (leaseId, details) =>
-      addEntry(KeyType.LeaseStatus, leaseId.arr)(booleanToBytes(details.isActive))
+      if (details.isActive)
+        addEntry(KeyType.LeaseStatus, leaseId.arr)(
+          booleanToBytes(true),
+          details.sender.arr,
+          details.recipient.bytes,
+          Longs.toByteArray(details.amount)
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/state/TxStateSnapshotHashBuilder.scala:84]: I see the [different](https://github.com/wavesplatform/protobuf-schemas/pull/84/files#diff-6cdd8c2ddb6ac7af1d937fa548cfcb0bd558c151ce202aa3895deb78a85d1cffR42-R45) order of fields: amount, sender, recipient.
Should we maintain the order of fields during the hashing? It looks like we haven't. Personally, the consistency is a good thing. What do you think?

**Review by vsuharnikov — APPROVED** [review]

File: node/src/main/scala/com/wavesplatform/state/StateSnapshot.scala [hunk 2]
```diff
@@ -172,23 +163,28 @@ object StateSnapshot {
...
+              height = 0
+            )
+          case _: TransactionStateSnapshot.LeaseState.Status.Cancelled | TransactionStateSnapshot.LeaseState.Status.Empty =>
+            ls.leaseId.toByteStr -> LeaseDetails(
+              PublicKey(ByteStr.fill(KeyLength)(0)),
+              Address(Array.fill(Address.HashLength)(0)),
+              0,
+              LeaseDetails.Status.Cancelled(0, None),
+              sourceId = ByteStr.empty,
+              height = 0
```
Comment (vsuharnikov) [inline node/src/main/scala/com/wavesplatform/state/StateSnapshot.scala]: As @phearnot said, we can fill the data here from the snapshot

File: node/src/main/scala/com/wavesplatform/state/TxStateSnapshotHashBuilder.scala (hunk 1 above)
Comment (xrtm000) [reply node/src/main/scala/com/wavesplatform/state/TxStateSnapshotHashBuilder.scala:84]: I think these cases are from different contexts

**Review by goodmai — APPROVED** [review]

**Review by ivan-mashonskiy — APPROVED** [review]

File: node/src/main/scala/com/wavesplatform/state/StateSnapshot.scala (hunk 2 above)
Comment (xrtm000) [reply node/src/main/scala/com/wavesplatform/state/StateSnapshot.scala]: Done

**Review by phearnot — APPROVED** [review]

**Pull request merged**